docs/src/configuration.md

```diff
@@ -52,26 +48,51 @@ with:
 $ mpiexecjl --project=/path/to/project -n 20 julia script.jl
 ```
 
-## Using a system-provided MPI
+## Using MPIPreferences.jl
+
+MPI.jl uses [Preferences.jl](https://github.com/JuliaPackaging/Preferences.jl) to
+allow the user to choose which MPI implementation to use for a project. This provides
+a single source of truth that can be used for `JLL` (Julia packages providing C libraries)
+that link against MPI, as well as localizing the choice of MPI implementation to a project.
+
+Users can use the provided [`use_system_binary`](@ref) or [`use_jll_binary`](@ref)
+to switch MPI implementations.
+
+### Migration from MPI.jl `v0.19`
+
+Prior to MPI.jl `v0.20` environment variables were used to configure which MPI
+library to use. These have now been removed.
 
-### Requirements
+- `JULIA_MPI_BINARY`
+- `JULIA_MPIEXEC`
+- `JULIA_MPI_INCLUDE_PATH`
+- `JULIA_MPI_CFLAGS`
+- `JULIA_MPICC`
+
+### Using a system-provided MPI
+
+#### Requirements
 
 MPI.jl requires a shared library installation of a C MPI library, supporting the MPI 3.0
 standard or later.
 
-### Building
+### Configuration
 
-To use the the system MPI, set the environment variable `JULIA_MPI_BINARY=system` and run
-`Pkg.build("MPI")`. This can be done by:
-```
-julia --project -e 'ENV["JULIA_MPI_BINARY"]="system"; using Pkg; Pkg.build("MPI"; verbose=true)'
+To use the the system MPI, run `MPI.use_system_binary()`.
+This will attempt find and identify any available MPI implementation, and create
+a file called `LocalPreferences.toml` adjacent to the current `Project.toml`.
+
+```sh
+julia --project -e 'using MPI; MPI.use_system_binary()'
 ```
-This will attempt find and identify any available MPI implementation.
+!!! note
+    You can copy `LocalPreferences.toml` to a different project, but you must list
+    `MPIPreferences` in the Project.toml either in `[deps]` or `[extras]` for the settings
+    to take effect. Due to a bug in Julia (until `v1.6.5` and `v1.7.1`), getting preferences
```

Comment:
Is `MPIPreferences` added automatically if `use_system_binary()` is called? If yes, this should be stated somewhere explicitly. Otherwise, it should me mentioned somewhere that this needs to be added manually.

It should also be stated somewhere that when switching back to the JLL binaries are used again, one can also safely delete the `LocalPreferences.jl` file. 

Also, be explicit about where it should be added: Pick either `[deps]` or `[extras]` and remove the burden of choice from the user. If it's all the same, I'd suggest `[deps]` since it's what is more well-known; one should only propagate adding it to `[extras]` if this is conceptually the "better" place.

Reply:
Yes MPIPreferences is added automatically to the `[extras]` section. Entries in `[deps]` need corresponding entries in `[compat]` (if one wants to register the package) and makes MPIPreferences implicitly loadable, so from my perspective `[extra]` is preferred.

Reply:
OK, in that case I suggest to rewrite this note to include this information, e.g.,
```suggestion
    You can copy `LocalPreferences.toml` to a different project, but you must list
    `MPIPreferences` in the Project.toml either in the `[extra]` section for the settings
    to take effect. Due to a bug in Julia (until `v1.6.5` and `v1.7.1`), getting preferences
```
but also state somewhere that `MPIPreferences` are automatically added to the Project.toml when `use_system_binary()` is called.